From 2a73f6439b6d745b2cbfff120dc7adc555400247 Mon Sep 17 00:00:00 2001
From: Michael Nosthoff <buildroot@heine.tech>
Date: Fri, 22 Jul 2022 08:49:48 +0200
Subject: [PATCH] disable unconditionally downloading api repos

The Buildroot build doesn't need those submodules, so disable them hard
for now.

See https://github.com/grpc/grpc/issues/30385

Signed-off-by: Michael Nosthoff <buildroot@heine.tech>
---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 94f8fd3e93..df43d2e193 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -402,6 +402,7 @@ if(gRPC_BUILD_GRPCPP_OTEL_PLUGIN)
   include(cmake/opentelemetry-cpp.cmake)
 endif()
 
+if(FALSE)
 # Setup external proto library at third_party/envoy-api with 2 download URLs
 if (NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/third_party/envoy-api AND gRPC_DOWNLOAD_ARCHIVES)
   # Download the archive via HTTP, validate the checksum, and extract to third_party/envoy-api.
@@ -478,6 +479,7 @@ if (NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/third_party/xds AND gRPC_DOWNLOAD_ARC
     xds-3a472e524827f72d1ad621c4983dd5af54c46776
   )
 endif()
+endif()
 
 if(WIN32)
   set(_gRPC_ALLTARGETS_LIBRARIES ${_gRPC_ALLTARGETS_LIBRARIES} ws2_32 crypt32)
-- 
2.34.1

